### PR TITLE
authorized_key: Add support for additional key types

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Fixes: #134

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/134_authorized_key.yml
plugins/modules/authorized_key.py

### CODE REVIEW

**Comment by Akasurde** [comment]
cc @quidame @saito-hideki @justjais Could you please review this? Thanks,

File: changelogs/fragments/134_authorized_key.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- authorized_key - add a list of valid key type (https://github.com/ansible-collections/ansible.posix/issues/134).
```
Comment (quidame) [inline changelogs/fragments/134_authorized_key.yml:2]:
```suggestion
- authorized_key - add a list of valid key types (https://github.com/ansible-collections/ansible.posix/issues/134).
```

**Review by quidame — APPROVED** [review]
@Akasurde thanks for improving this module !

**Review by saito-hideki — APPROVED** [review]
@Akasurde Thank you for the PR! I have tested `VALID_SSH2_KEY_TYPES` and your changes are working fine on my testing environment. :)

**Comment by quidame** [comment]
the CI tests error doesn't seem to be related to this PR

**Comment by Akasurde** [comment]
Closing and re-opening for CI trigger.

**Pull request merged**